datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -1478,6 +1478,28 @@ macro_rules! to_string {
     }};
 }
 
+
+/// Array_union SQL function
+pub fn array_union(args: &[ArrayRef]) -> Result<ArrayRef> {
+    let array1 = &args[0];
+    let array2= &args[1];
+
+    check_datatypes("array_union", &[array1, array2])?;
+    let list1 = as_list_array(array1)?;
```

Comment:
Wondering can we reuse `.data_type()` on `array1` level so we can get rid of unneccessary conversion

Reply:
this is my first attempt to contribute to arrow-datafusion, can you explain why we are converting a generic array to i32?

Reply:
@comphead I made one more iteration, but I am still learning the API. where should I look to find a way to concatenate two primitive types array? `concat_internal` doesn't work